### ISSUE TITLE
RESTEasy fails to build native image on latest 17 build of GraalVM Community  and GraalVM

### ISSUE THREAD

**Issue by Christopher-Chianelli** [quarkusio/quarkus#44877]
### Describe the bug

In our build matrix, we use the `setup-graalvm` action to get the latest GraalVM  corresponding to the matrix's Java version:

```
  - uses: graalvm/setup-graalvm@v1
    with:
      java-version: ${{matrix.java-version}}
      distribution: 'graalvm-community'
      components: 'native-image'
      github-token: ${{ secrets.GITHUB_TOKEN }}
      cache: 'maven'
```

For Java 17, this fetches  `java 17.0.9-graalce` (which can be installed with  SDKMan: `sdk install java 17.0.9-graalce`)

RESTEasy uses Netty, which causes a  build time error:

```
Error: Classes that should be initialized at run time got initialized during image building:
 io.netty.buffer.UnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeDirectByteBuf
```

The issue does not reproduce using latest Java 17 version of Mandrel (23.0.6.r17-mandrel).

### Expected behavior

The `REST` sample project generated by code.quarkus.io native image builds successfully on the latest version of GraalVM for Java 17.

### Actual behavior

This error appears during the native image build:

```
Error: Classes that should be initialized at run time got initialized during image building:
 io.netty.buffer.UnpooledUnsafeDirectByteBuf the class was requested to be initialized at run time (subtype of io.netty.buffer.AbstractReferenceCountedByteBuf). To see why io.netty.buffer.UnpooledUnsafeDirectByteBuf got initialized use --trace-class-initialization=io.netty.buffer.UnpooledUnsafeDirectByteBuf
```

### How to Reproduce?

Reproducer: https://github.com/Christopher-Chianelli/issue-reproducer/tree/quarkus-native-17-failed-build
(App was generated from code.quarkus.io)

### Output of `uname -a` or `ver`

Linux 6.11.10-300.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Nov 23 00:51:20 UTC 2024 x86_64 GNU/Linux

### Output of `java -version`

Java(TM) SE Runtime Environment Oracle GraalVM 17.0.12+8.1 (build 17.0.12+8-LTS-jvmci-23.0-b41)

### Quarkus version or git rev

3.17.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

3.9.9

**Comment by gsmet** [comment]
Yeah, we had all sorts of weird issues with the Java 17 version of GraalVM. Even if you app is using Java 17, we recommend to use the Java 21 one.

We need to find a place to document it properly.
